dataset.go

```
@@ -36,6 +30,15 @@ type Dataset interface {
 	Split(secondWidth int) (Dataset, Dataset)
 }
 
+type datasetType struct{}
+
+func (sett *datasetType) String() string  { return sett.Name() }
+func (sett *datasetType) Name() string    { return "record" }
+func (sett *datasetType) Data(n int) Data { return sett.DataEmpty(n) }
```

why didn't you return here an error with panic instead of using DataEmpty?

---

just for not repeating the error message. they both should fail of corse...